client_plugin/drivers/shared/kvstore/etcdops/etcdops.go

```diff
@@ -338,12 +346,44 @@ func (e *EtcdKVS) etcdEventHandler(ev *etcdClient.Event) {
 			return
 		}
 
-		if fn(volName) {
-			// server start/stop succeed, set desired state
-			if e.CompareAndPut(kvstore.VolPrefixState+volName,
+		port, servName, succeeded := fn(volName)
+		if succeeded {
+			// Update volume metadata to reflect
+			// port number and file service name
+			var entries []kvstore.KvPair
+			var volRecord sharedVolConnectivityData
+			volRecord.Port = port
+			volRecord.ServiceName = servName
+			byteRecord, err := json.Marshal(volRecord)
+			if err != nil {
+				log.Warningf("Failed to update port and file service name: %v",
+					err)
+				e.CompareAndPut(kvstore.VolPrefixState+volName,
```

Comment:
Please add a comment for this line to explain what is this line doing. It will help others to understand the code.